youtube_dl/extractor/seeso.py

```diff
@@ -0,0 +1,91 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+import urlparse
+import os.path
+import json
+from ..utils import (
+    sanitized_Request,
+    urlencode_postdata
+)
+
+
+class SeesoIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?seeso\.com/view/episode/(?P<id>[0-9]+)'
+    _TEST = {
+        'params': {
+            'username': 'emailhere',
+            'password': 'passwordhere'
+        },
+        'url': 'https://www.seeso.com/view/episode/799241283849',
+        'info_dict': {
+            'id': '799241283849',
+            'ext': 'mp4',
+            'series': 'Bajillion Dollar Propertie$',
+            'title': 'Farsi Lessons',
+            'description': ("Amir leads Victoria into a trap, the Bros meet a super obnoxious bro, "
+                            "and rival brokers Serge and Gio intimidate Glenn."),
+            'thumbnail': 'https://chaosic.akamaized.net/NBCOTT_-_Production/360/907/160831_3092487_Farsi_Lessons.jpg',
+        }
+    }
+
+    def _real_extract(self, url):
+        username, password = self._get_login_info()
+        video_id = self._match_id(url)
+
+        if username is None or password is None:
+            return
+
+        _API_AUTH_URL = "https://www.seeso.com/api/v3/auth/login"
+        auth_json = {
+            "username": username,
+            "password": password,
+            "userOnly": True,
+            "isRememberMe": True,
+            "isWeb": True
+        }
+
+        # Send auth POST request and get token from response
+        auth_request = sanitized_Request(_API_AUTH_URL, urlencode_postdata(auth_json))
+        auth_response = json.loads(self._download_webpage(auth_request, '', note='Getting auth token...'))
```

Comment:
I had created a _login method before, but I couldn't for the life of me get the extract method to call it. I'll take another run at it.